central/securedclustercertgen/certificates.go

```diff
@@ -13,33 +13,64 @@ import (
+var scannerV4ServiceTypes = set.NewFrozenSet[storage.ServiceType](storage.ServiceType_SCANNER_V4_INDEXER_SERVICE, storage.ServiceType_SCANNER_V4_DB_SERVICE)
+var localScannerServiceTypes = func() set.FrozenSet[storage.ServiceType] {
+	types := scannerV2ServiceTypes.Unfreeze()
+	types = types.Union(scannerV4ServiceTypes.Unfreeze())
+	return types.Freeze()
+}()
+
+var securedClusterServiceTypes = set.NewFrozenSet[storage.ServiceType](
+	storage.ServiceType_SENSOR_SERVICE,
+	storage.ServiceType_COLLECTOR_SERVICE,
+	storage.ServiceType_ADMISSION_CONTROL_SERVICE)
+
 var allSupportedServiceTypes = func() set.FrozenSet[storage.ServiceType] {
-	types := v2ServiceTypes.Unfreeze()
-	types = types.Union(v4ServiceTypes.Unfreeze())
+	types := securedClusterServiceTypes.Unfreeze()
+	types = types.Union(localScannerServiceTypes.Unfreeze())
 	return types.Freeze()
 }()
 
+type certIssuerImpl struct {
+	allSupportedServiceTypes set.FrozenSet[storage.ServiceType]
+}
+
+// IssueSecuredClusterCerts issues certificates for all the services of a secured cluster (including local scanner).
+func IssueSecuredClusterCerts(namespace string, clusterID string) (*storage.TypedServiceCertificateSet, error) {
+	certIssuer := certIssuerImpl{
+		allSupportedServiceTypes: allSupportedServiceTypes,
+	}
+	return certIssuer.issueCertificates(allSupportedServiceTypes, namespace, clusterID)
+}
+
 // IssueLocalScannerCerts issue certificates for a local scanner running in secured clusters.
 func IssueLocalScannerCerts(namespace string, clusterID string) (*storage.TypedServiceCertificateSet, error) {
+	// In any case, generate certificates for Scanner v2.
+	serviceTypes := scannerV2ServiceTypes
+	if features.ScannerV4.Enabled() {
+		// Additionally, generate certificates for Scanner v4.
+		serviceTypes = localScannerServiceTypes
+	}
+
+	certIssuer := certIssuerImpl{
+		allSupportedServiceTypes: localScannerServiceTypes,
+	}
+
+	return certIssuer.issueCertificates(serviceTypes, namespace, clusterID)
+}
+
+func (c *certIssuerImpl) issueCertificates(serviceTypes set.FrozenSet[storage.ServiceType], namespace string, clusterID string) (*storage.TypedServiceCertificateSet, error) {
 	if namespace == "" {
-		return nil, errors.New("namespace is required to issue the certificates for the local scanner")
+		return nil, errors.New("namespace is required to issue the certificates for the secured cluster")
 	}
 
 	var certIssueError error
 	var caPem []byte
 
-	// In any case, generate certificates for Scanner v2.
-	serviceTypes := v2ServiceTypes
-	if features.ScannerV4.Enabled() {
-		// Additionally, generate certificates for Scanner v4.
-		serviceTypes = allSupportedServiceTypes
-	}
-
 	serviceCerts := make([]*storage.TypedServiceCertificate, 0, serviceTypes.Cardinality())
 	for _, serviceType := range serviceTypes.AsSlice() {
-		ca, cert, err := localScannerCertificatesFor(serviceType, namespace, clusterID)
+		ca, cert, err := c.certificatesFor(serviceType, namespace, clusterID)
```

Comment:
Where do we throw them away, on the Sensor side? I don't want to change the behaviour of `IssueLocalScannerCerts` in this PR. And sending CA certs will be needed in the future, when we implement central CA cert auto renewal.

Reply:
With "throwing away" I mean the following: In every iteration of the loop we are retrieving the CA, since it is being returned by `c.certificatesFor` and we assign it to `ca`. But we are not using except in the first iteration of the loop, where we assign it to `caPem`, and end up returning it to the caller. Not a big deal, just an indicator for me that the internal API might be a lacking a better CA handling and a function for just returning new certificates, not the CA.

Reply:
Feel free to close.

Reply:
I understand now what you mean, but I still wouldn't change this in this PR, so I'll mark this as resolved if you don't mind.